### PR TITLE
fix a bug where keys are not properly translated to volume id

### PR DESCRIPTION
**Pull request by hantaowang**
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Flaky tests

**Special notes for your reviewer**:
@davidz627 @saad-ali 

**Does this PR introduce a user-facing change?**:
```release-note
Fixed a bug that causes disks in the same zone/region to be provisioned serially
```

### CODE REVIEW

File: pkg/common/utils.go [hunk 1]
```diff
@@ -76,9 +76,9 @@ func VolumeIDToKey(id string) (*meta.Key, error) {
 func KeyToVolumeID(volKey *meta.Key, project string) (string, error) {
 	switch volKey.Type() {
 	case meta.Zonal:
-		return fmt.Sprintf(volIDZonalFmt, project, volKey.Zone, volKey.Zone), nil
+		return fmt.Sprintf(volIDZonalFmt, project, volKey.Zone, volKey.Name), nil
```
Comment (davidz627) [inline pkg/common/utils.go]: please add unit test for `KeyToVolumeID`. We should have done that when adding the function in the first place, oops

**Comment by hantaowang** [comment]
@davidz627 @jingxu97 ready for review thanks!

**Pull request merged**